### PR TITLE
bump version and changelog for 1.12.4-rc1

### PR DESCRIPTION
**Pull request by vieux**
ping @andrewhsu @aaronlehmann @mavenugo @dmcgowan @tonistiigi

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,83 @@ information on the list of deprecated flags and APIs please have a look at
...
+systemctl restart docker` to reload changes and (re)start the docker daemon.
+
+
+### Runtime
+
+- Fix issue where volume metadata was not removed [#29083](https://github.com/docker/docker/pull/29083)
+- Asynchronously close streams to prevent holding container lock [#29050](https://github.com/docker/docker/pull/29050)
+- Relabel newly created container volumes [#29050](https://github.com/docker/docker/pull/29050)
+- Remove hostname validation [#28990](https://github.com/docker/docker/pull/28990)
+- Various IO fixes [#29095](https://github.com/docker/docker/pull/29095)
```
Comment (tonistiigi) [inline CHANGELOG.md]: Fix deadlocks caused by IO races

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,83 @@ information on the list of deprecated flags and APIs please have a look at
...
+
+After making those changes, run `sudo systemctl daemon-reload`, and `sudo
+systemctl restart docker` to reload changes and (re)start the docker daemon.
+
+
+### Runtime
+
+- Fix issue where volume metadata was not removed [#29083](https://github.com/docker/docker/pull/29083)
+- Asynchronously close streams to prevent holding container lock [#29050](https://github.com/docker/docker/pull/29050)
+- Relabel newly created container volumes [#29050](https://github.com/docker/docker/pull/29050)
```
Comment (cpuguy83) [inline CHANGELOG.md]: Fix selinux labels for newly created container volumes

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,83 @@ information on the list of deprecated flags and APIs please have a look at
...
+  - Fix a panic in libnetwork.(*sandbox).execFunc [docker/libnetwork#1556](https://github.com/docker/libnetwork/pull/1556)
+  - Honor icc=false for internal networks [docker/libnetwork#1525](https://github.com/docker/libnetwork/pull/1525)
+
+### Logging
+
+* Update syslog log driver [#29150](https://github.com/docker/docker/pull/29150)
+
+### Distribution
+
+- Ensure opaque directory permissions respected [#29093](https://github.com/docker/docker/pull/29093)
```
Comment (dmcgowan) [inline CHANGELOG.md]: "Fix incorrect opaque directory permission in overlay2"

Also maybe in Runtime instead of distribution?

Comment (vieux) [inline CHANGELOG.md]: updated

**Comment by tonistiigi** [comment]
LGTM

**Comment by dmcgowan** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,80 @@ information on the list of deprecated flags and APIs please have a look at
 https://docs.docker.com/engine/deprecated/ where target removal dates can also
 be found.
 
+## 1.12.4
```
Comment (andrewhsu) [inline CHANGELOG.md]: @vieux missing date `(2016-12-06)`

which is causing build errors of rpm pkgs:
```
error: %changelog entries must start with *
```

**Pull request merged**